Major overhaul of provider init and Umami Provider (#492)
patternfly/chatbot: packages/module/patternfly-docs/content/extensions/chatbot/examples/Analytics/Analytics.md

### Setup

> 1. Before you can use the `trackingAPI`, you must first supply the API keys of the respective providers. Providers are enabled by adding them to the `activeProviders` property:

Reviewer: ```suggestion
1. Before you can use the `trackingAPI`, you must first supply the API keys of the respective providers. To enable a provider, it must be added to the `activeProviders` property:
```

        },

  posthogKey: 'TODO-key',
  umamiKey: 'TODO-umami-key',
  umamiHostUrl: 'http://localhost:3000', // TODO where is your JS provider?
  'umami-data-domains': 'TODO umami data domain',
  something: 'test',
};
```

> Note, that there is also a key `verbose` that allows you to enable debugging output on web-browser console. By default, this is set to `false`.

Reviewer: ```suggestion
- **Note:** To enable output debugging via the web-browser console, set the `verbose` key to `true`. By default, this is set to `false`.
```
I'm assuming that the key must be true, but if that's wrong then we can leave that part off

#### Tracking providers

Only providers with a matching entry in the `InitProps.activeProviders` array will be started and used.

Possible values are:
* Umami
* Posthog
* Segment
* Console

##### Modifying providers

If you know upfront that you only want to use 1 of the supported providers, you can modify `getTrackingProviders()` and remove all other providers in the providers array.

When using the providers, you might need to add additional dependencies to your package.json file:

```nolive
"dependencies": {
  "@segment/analytics-next": "^1.76.0",
  "posthog-js": "^1.194.4"
```

Depending on your local setup, this might not be necessary. For example, if you pull the ChatBot codebase as a dependency into your project, you don't need to add it as an additional dependency in your package.json.

##### Adding providers

To add another analytics provider, you need to implement 2 interfaces, `TrackingSpi` and `trackingApi`.

1. It is easiest to start by copying the `ConsoleTrackingProvider`

> 1. The first thing you should do is to add it to the `Providers` enum in `tracking_spi.ts`

Reviewer: ```suggestion
1. Add an entry for your new provider to the `Providers` enum in `tracking_spi.ts`
```

Could we update this? Just to avoid the use of "it", which reads a little ambiguous

#### Tracking providers

Only providers with a matching entry in the `InitProps.activeProviders` array will be started and used.

Possible values are:
* Umami
* Posthog
* Segment
* Console

##### Modifying providers

If you know upfront that you only want to use 1 of the supported providers, you can modify `getTrackingProviders()` and remove all other providers in the providers array.

> When using the providers you may need to add additional dependencies to your package.json file:

Reviewer: ```suggestion
When using the providers, you might need to add additional dependencies to your package.json file:
```

```nolive
"dependencies": {
  "@segment/analytics-next": "^1.76.0",
  "posthog-js": "^1.194.4"
```

> This depends on your local setup. If you pull in
> the chatbot codebase as dependency into your project, you are already set. 
> 

Reviewer: ```suggestion
Depending on your local setup, this might not be necessary. For example, if you pull the ChatBot codebase as a dependency into your project, you don't need to add it as an additional dependency in your package.json.

```

##### Adding providers

To add another analytics provider, you need to implement 2 interfaces, `TrackingSpi` and `trackingApi`.

1. It is easiest to start by copying the `ConsoleTrackingProvider`

> 1. The first thing you should do is to add it to the `Providers` enum in `tracking_spi.ts`

Author: This is not right. What I meant was to add an entry for the new provider to the enum.

> 1. The first thing you should do is to add it to the `Providers` enum in `tracking_spi.ts`

Reviewer: Ah I see, like this?
1. Add an entry for your new provider to the `Providers` enum in `tracking_spi.ts`
